src/atomate2/vasp/jobs/adsorption.py

```diff
@@ -0,0 +1,460 @@
+    adslabs_data: dict[str, list],
+    molecule_dft_energy: float,
+    slab_dft_energy: float,
+) -> dict[str, list[Any]]:
+    """Calculate the adsorption energies by subtracting the energies of
+    the adsorbate, slab, and adsorbate-slab.
+
+    Parameters
+    ----------
+    bulk_structure: Structure
+        The bulk/unit cell structure.
+    molecule_structure: Structure
+        The molecule to be adsorbed.
+    surface_idx: tuple
+        The Miller index [h, k, l] of the surface.
+    adslab_structures: list[Structure]
+        The list of all possible configurations of slab structures with adsorbates.
+    ads_outputs: dict[str, list]
+        The outputs of the adsorption calculations.
+    molecule_dft_energy: float
+        The static energy of the molecule.
+    slab_dft_energy: float
+        The static energy of the slab.
+
+    Returns
+    -------
+    list
+        The list of (adsorption configurations, configuration number,
+        adsorption energy, directories) sorted by adsorption energy.
+    """  # noqa: D205
+    outputs: dict[str, list] = {
+        "adsorption_configuration": [],
+        "configuration_number": [],
+        "adsorption_energy": [],
+        "dirs": [],
+    }
+
+    for i, ad_structure in enumerate(adslab_structures):
+        outputs["adsorption_configuration"].append(ad_structure)
+        outputs["configuration_number"].append(i)
+        ads_energy = (
+            adslabs_data["static_energy"][i] - molecule_dft_energy - slab_dft_energy
+        )
+        outputs["adsorption_energy"].append(ads_energy)
+        outputs["dirs"].append(adslabs_data["dirs"][i])
+
+    # Combine and sort
+    combined_outputs = list(
+        zip(
+            outputs["adsorption_configuration"],
+            outputs["configuration_number"],
+            outputs["adsorption_energy"],
+            outputs["dirs"],
+        )
+    )
+    sorted_combined_outputs = sorted(combined_outputs, key=lambda x: x[2])
+
+    return {
+        "adsorption_configuration": [item[0] for item in sorted_combined_outputs],
+        "configuration_number": [item[1] for item in sorted_combined_outputs],
+        "adsorption_energy": [item[2] for item in sorted_combined_outputs],
+        "dirs": [item[3] for item in sorted_combined_outputs],
+    }
+
+
+@dataclass
+class BulkRelaxMaker(BaseVaspMaker):
+    """Maker for molecule relaxation.
+
+    Parameters
+    ----------
+    name: str
+        The name of the flow produced by the maker.
+    input_set_generator: VaspInputGenerator
+        The input set generator for the relaxation calculation.
+    """
+
+    name: str = "bulk relaxation maker - "
+    input_set_generator: VaspInputGenerator = field(
+        default_factory=lambda: RelaxSetGenerator(
+            user_kpoints_settings=Kpoints.from_dict(
+                {
+                    "nkpoints": 0,
+                    "generation_style": "Gamma",
+                    "kpoints": [[11, 11, 11]],
```

Comment:
Is fixing the Kpoints settings a good idea? A better approach would be to set a k-point density. E.g., `user_kpoints_settings={"reciprocal_density": 200}`. This would then work with systems that aren't cubic.

Reply:
Yeah I think you are right! I will update the setting to k-point density for the default option. Thanks for the suggestion!